Fails to cross-compile with xgo using sqlite_vtable

Hi there! I'm having an issue cross-compiling a project that depends on this package. I'm using `xgo` from a mac and the virtual table implementation, but running a command like so:

`xgo -tags=sqlite_vtable ...`

I get a failure that looks like:

```
# github.com/mattn/go-sqlite3
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:292:13: type [1073741824]_Ctype_struct_sqlite3_index_constraint larger than address space
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:292:13: type [1073741824]_Ctype_struct_sqlite3_index_constraint too large
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:311:13: type [1073741824]_Ctype_struct_sqlite3_index_orderby larger than address space
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:311:13: type [1073741824]_Ctype_struct_sqlite3_index_orderby too large
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:422:9: type [1073741824]_Ctype_struct_sqlite3_index_constraint_usage larger than address space
/go/pkg/mod/github.com/mattn/go-sqlite3@v2.0.3+incompatible/sqlite3_opt_vtable.go:422:9: type [1073741824]_Ctype_struct_sqlite3_index_constraint_usage too large
2020/08/09 19:07:09 Failed to cross compile package: exit status 2.
```

Any pointers in the right direction would be much appreciated! Thanks so much

--- a/sqlite3_opt_vtable.go
+++ b/sqlite3_opt_vtable.go
@@ -288,10 +288,13 @@ type InfoOrderBy struct {
 }
 
 func constraints(info *C.sqlite3_index_info) []InfoConstraint {
-	l := info.nConstraint
-	slice := (*[1 << 30]C.struct_sqlite3_index_constraint)(unsafe.Pointer(info.aConstraint))[:l:l]
+	slice := *(*[]C.struct_sqlite3_index_constraint)(unsafe.Pointer(&reflect.SliceHeader{
+		Data: uintptr(unsafe.Pointer(info.aConstraint)),
+		Len:  int(info.nConstraint),
+		Cap:  int(info.nConstraint),
+	}))
 
-	cst := make([]InfoConstraint, 0, l)
+	cst := make([]InfoConstraint, 0, len(slice))
 	for _, c := range slice {
 		var usable bool
 		if c.usable > 0 {
@@ -307,10 +310,13 @@ func constraints(info *C.sqlite3_index_info) []InfoConstraint {
 }
 
 func orderBys(info *C.sqlite3_index_info) []InfoOrderBy {
-	l := info.nOrderBy
-	slice := (*[1 << 30]C.struct_sqlite3_index_orderby)(unsafe.Pointer(info.aOrderBy))[:l:l]
+	slice := *(*[]C.struct_sqlite3_index_orderby)(unsafe.Pointer(&reflect.SliceHeader{
+		Data: uintptr(unsafe.Pointer(info.aOrderBy)),
+		Len:  int(info.nOrderBy),
+		Cap:  int(info.nOrderBy),
+	}))
 
-	ob := make([]InfoOrderBy, 0, l)
+	ob := make([]InfoOrderBy, 0, len(slice))
 	for _, c := range slice {
 		var desc bool
 		if c.desc > 0 {
@@ -418,13 +424,17 @@ func goVBestIndex(pVTab unsafe.Pointer, icp unsafe.Pointer) *C.char {
 	}
 
 	// Get a pointer to constraint_usage struct so we can update in place.
-	l := info.nConstraint
-	s := (*[1 << 30]C.struct_sqlite3_index_constraint_usage)(unsafe.Pointer(info.aConstraintUsage))[:l:l]
+
+	slice := *(*[]C.struct_sqlite3_index_constraint_usage)(unsafe.Pointer(&reflect.SliceHeader{
+		Data: uintptr(unsafe.Pointer(info.aConstraintUsage)),
+		Len:  int(info.nConstraint),
+		Cap:  int(info.nConstraint),
+	}))
 	index := 1
-	for i := C.int(0); i < info.nConstraint; i++ {
+	for i := range slice {
 		if res.Used[i] {
-			s[i].argvIndex = C.int(index)
-			s[i].omit = C.uchar(1)
+			slice[i].argvIndex = C.int(index)
+			slice[i].omit = C.uchar(1)
 			index++
 		}
 	}

